addons/website/static/src/builder/website_builder.js

```diff
@@ -91,6 +91,15 @@ export class WebsiteBuilder extends Component {
     }
 
     discard() {
+        const dispatchDiscardAndCloseEditor = () => {
+            if (this.editor) {
+                for (const handler of this.editor.getResource("discard_handlers") || []) {
+                    handler();
+                }
+            }
+            this.props.builderProps.closeEditor();
+        };
```

Comment:
When you introduce a "feature" / a "function, it is generally better to include it in the commit that actually uses it IMO. So this can be squashed with the other commit.